Modules/Quest/QuestEventHandler.lua

```diff
@@ -0,0 +1,352 @@
+local MAX_QUEST_LOG_INDEX = 75
+
+local QUEST_LOG_STATES = {
+    QUEST_ACCEPTED = "QUEST_ACCEPTED",
+    QUEST_TURNED_IN = "QUEST_TURNED_IN",
+    QUEST_REMOVED = "QUEST_REMOVED",
+    QUEST_ABANDONED = "QUEST_ABANDONED"
+}
+local MAX_OBJECTIVE_TEXT_TRIES = 5
+
+local eventFrame = CreateFrame("Frame", "QuestieQuestEventFrame")
+local questLog = {}
+local questLogUpdateQueueSize = 1
+local skipNextUQLCEvent = false
+local doFullQuestLogScan = false
+
+
+--- Registers all events that are required for questing (accepting, removing, objective updates, ...)
+function QuestEventHandler:RegisterEvents()
+    print("[Quest Event] RegisterEvents")
+    eventFrame:RegisterEvent("QUEST_ACCEPTED")
+    eventFrame:RegisterEvent("QUEST_TURNED_IN")
+    eventFrame:RegisterEvent("QUEST_REMOVED")
+    eventFrame:RegisterEvent("QUEST_LOG_UPDATE")
+    eventFrame:RegisterEvent("QUEST_WATCH_UPDATE")
+    eventFrame:RegisterEvent("UNIT_QUEST_LOG_CHANGED")
+    eventFrame:RegisterEvent("BANKFRAME_CLOSED")
+    eventFrame:RegisterEvent("UNIT_FACTION")
+    eventFrame:SetScript("OnEvent", _QuestEventHandler.OnEvent)
+
+    _QuestEventHandler:InitQuestLog()
+end
+
+--- On Login mark all quests in the quest log with QUEST_ACCEPTED state
+function _QuestEventHandler:InitQuestLog()
+    for i = 1, MAX_QUEST_LOG_INDEX do
+        local title, _, _, isHeader, _, _, _, questId = GetQuestLogTitle(i)
+        if (not title) then
+            break -- We exceeded the valid quest log entries
+        end
+        if (not isHeader) then
+            questLog[questId] = {
+                state = QUEST_LOG_STATES.QUEST_ACCEPTED
+            }
+            QuestieLib:CacheItemNames(questId)
+        end
+    end
+
+    QuestieHash:InitQuestLogHashes()
+end
+
+--- Fires when a quest is accepted in anyway.
+---@param questLogIndex number
+---@param questId number
+function _QuestEventHandler:QuestAccepted(questLogIndex, questId)
+    print("[Quest Event] QUEST_ACCEPTED", questLogIndex, questId)
+
+    if questLog[questId] and questLog[questId].timer then
+        -- We had a QUEST_REMOVED event which started this timer and now it was accepted again.
+        -- So the quest was abandoned before, because QUEST_TURNED_IN would have run before QUEST_ACCEPTED.
+        questLog[questId].timer:Cancel()
+        questLog[questId].timer = nil
+        _QuestEventHandler:MarkQuestAsAbandoned(questId)
+    end
+
+    questLog[questId] = {}
+    skipNextUQLCEvent = true
+    QuestieLib:CacheItemNames(questId)
+    _QuestEventHandler:HandleQuestAccepted(questId)
+end
+
+local objectiveTextTries = 0
+---@param questId number
+---@return boolean true @if the function was successful, false otherwise
+function _QuestEventHandler:HandleQuestAccepted(questId)
+    if objectiveTextTries == MAX_OBJECTIVE_TEXT_TRIES then
+        -- Check for failing recursion. Something is up if the objective texts are still invalid.
```

Comment:
Only single counter is used for all quests that are possibly in accept process at the same time and the counter is not zeroed after fail.

Also after more experience with game current behavior, I would maybe just remove whole "failing recursion" check.
Maybe just add some timer for if the objectives aren't good even in 15 seconds print message for user to report the case. Do NOT do anything else in timer than print message.